src/core/qgstextrenderer.h

```diff
@@ -1022,6 +1022,7 @@ class CORE_EXPORT QgsTextFormat
     {
       HorizontalOrientation, //!< Vertically oriented text
       VerticalOrientation, //!< Horizontally oriented text
+      RotationBasedOrientation, //!< Horizontally or vertically oriented text based on rotation
```

Comment:
Better note here that it's only usable during map labelling, not in standalone text rendering